packages/react-dom/src/events/__tests__/SyntheticKeyboardEvent-test.js

```diff
@@ -9,114 +9,328 @@
 
 'use strict';
 
-var SyntheticKeyboardEvent;
-var getEventCharCode;
+let React;
+let ReactDOM;
+
+let getEventCharCode;
 
 describe('SyntheticKeyboardEvent', () => {
-  var createEvent;
+  let container;
+
+  beforeEach(() => {    
+    React = require('react');
+    ReactDOM = require('react-dom');
+    // The container has to be attached for events to fire.
+    container = document.createElement('div');
+    document.body.appendChild(container);
 
-  beforeEach(() => {
-    // Mock getEventCharCode for proper unit testing
-    jest.mock('../getEventCharCode');
     getEventCharCode = require('../getEventCharCode').default;
 
-    // TODO: can we express this test with only public API?
-    SyntheticKeyboardEvent = require('../SyntheticKeyboardEvent').default;
-    createEvent = function(nativeEvent) {
-      var target = require('../getEventTarget').default(nativeEvent);
-      return SyntheticKeyboardEvent.getPooled({}, '', nativeEvent, target);
-    };
   });
 
+  afterEach(() => {
+    document.body.removeChild(container);
+    container = null;
+  });
+
+
   describe('KeyboardEvent interface', () => {
     describe('charCode', () => {
       describe('when event is `keypress`', () => {
         it('returns whatever getEventCharCode returns', () => {
```

Comment:
A lot of these tests were written in a very unit testing centric way. You need to try to liberate from that mindset. Think what's actually *valuable* to test for from perspective of library user. Not all the possible inputs to internal functions that never get called with those inputs in the first place.